### PR TITLE
Reject p < 1 in triplet_margin_loss to prevent NaN from p-norm overflow

### LINKED ISSUE #187178: torch.nn.functional.triplet_margin_loss returns NaN for p < 1 (p-norm overflow)
### 🐛 Describe the bug

`torch.nn.functional.triplet_margin_loss(anchor, positive, negative, p)` computes a p-norm distance `d(a, b) = ((|a - b| + eps)^p).sum(-1)^(1/p)` for both the positive and negative pairs. For very small `p` each `(|a - b| + eps)^p ~= 1`, so the per-row sum is `~= N` (the feature dim) and `N^(1/p)` overflows fp32 to `+inf` once `(1/p) * log(N)` exceeds `log(3.4e38) ~= 88.7`. The loss is then `d_ap - d_an + margin = inf - inf + margin = NaN`, with no exception.

The class docstring annotates `p (int, optional)`, implying `p >= 1`, but the function silently accepts arbitrary floats and never validates `p`. `torch.nn.TripletMarginLoss(p=1e-6)` is affected identically. This is the small-`p` companion of the large-`p` `+inf` overflow in `torch.nn.PairwiseDistance` (#184036): both come from evaluating the p-norm power directly instead of in log space.

```python
import torch
import torch.nn.functional as F

torch.manual_seed(0)
anchor   = torch.randn(4, 8)
positive = torch.randn(4, 8)
negative = torch.randn(4, 8)
loss = F.triplet_margin_loss(anchor, positive, negative, margin=1.0, p=1e-6)
print(loss)
```

Observed:

```
tensor(nan)
```

Expected:

Either (a) raise `ValueError` for `p < 1` (or `p <= 0`), matching the documented `p (int)` contract, or (b) evaluate the p-norm in log space via `exp((1/p) * logsumexp(p * log(|a - b| + eps)))`, which stays finite for any `p > 0`.

### Versions

```
PyTorch version: 2.5.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-124-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1660 SUPER
Nvidia driver version: 535.309.01
cuDNN version: Could not collect
Is XNNPACK available: True

CPU:
Architecture:                            x86_64
CPU(s):                                  20
Model name:                              12th Gen Intel(R) Core(TM) i7-12700
Thread(s) per core:                      2
Core(s) per socket:                      12

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] torch==2.5.1+cu121
[pip3] triton==3.1.0
```

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by rockygeekz**
## Summary

`F.triplet_margin_loss` silently returns NaN when `p < 1` because the p-norm computation overflows. For very small p, each `(|a - b| + eps)^p` is close to 1, so the sum across the feature dimension is roughly N, and `N^(1/p)` blows past fp32 range. The loss then becomes `inf - inf + margin = NaN` with no error.

The docstring already says `p (int, optional)` implying p >= 1, but nothing enforced it. This adds a `ValueError` for `p < 1` in both `F.triplet_margin_loss` and `nn.TripletMarginLoss`, matching the existing `margin > 0` validation pattern.

**Note:** Reproduced the issue and verified the fix locally on torch 2.12.0+cpu.

## Test plan

Added `test_triplet_margin_loss_invalid_p` covering:
  - `p=0.5` raises ValueError (functional)
  - `p=1e-6` raises ValueError (functional)
  - `p=0.5` raises ValueError (module)
  - `p=-1.0` raises ValueError (module)
  - `p=1` and `p=2` still produce valid (non NaN) loss

```
python test/test_nn.py TestNN.test_triplet_margin_loss_invalid_p
python test/test_nn.py TestNN.test_triplet_margin_loss
```

Fixes #187178

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Thanks

**Comment by albanD** [comment]
Can you please rebase on the latest main branch?
